NitroxPatcher/Patches/ItemsContainer_NotifyRemoveItem_Patch.cs

```diff
@@ -13,9 +13,13 @@ public class ItemsContainer_NotifyRemoveItem_Patch : NitroxPatch
 
         public static void Postfix(ItemsContainer __instance, InventoryItem item)
         {
+            LocalPlayer localPlayer = NitroxServiceLocator.LocateService<LocalPlayer>();
             if (item != null && __instance.tr.parent.name != "EscapePod" && __instance.tr.parent.name != "Player")
             {
-                NitroxServiceLocator.LocateService<ItemContainers>().BroadcastItemRemoval(item.item, __instance.tr.parent.gameObject);
+                if (!localPlayer.Shutdown)
```

Comment:
Do we know why this code path is being called when the client disconnects?  Would it be better to patch the code path that triggers this to not trigger this in multiplayer?

Reply:
if we path the function make that bug on user go to main screen the client dont release memory resources from current session i   find a better file for catch the event work went user press Alt+F4 to

 
namespace NitroxPatcher.Patches
{
    public class Player_OnDestroy_Patch : NitroxPatch
    {
        public static readonly Type TARGET_CLASS = typeof(Player);
        public static readonly MethodInfo TARGET_METHOD = TARGET_CLASS.GetMethod("OnDestroy", BindingFlags.NonPublic | BindingFlags.Instance);

        public static void Prefix()
        {
            LocalPlayer localPlayer = NitroxServiceLocator.LocateService<LocalPlayer>();
            if (localPlayer != null)
            {
                localPlayer.Shutdown = true;
            }
            Log.Info("Player OnDestroyEvent");
        }

        public override void Patch(HarmonyInstance harmony)
        {
            PatchPrefix(harmony, TARGET_METHOD);
        }
    }
}